client/plan_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client_test
+
+import (
+	"encoding/json"
+	"net/url"
+
+	"github.com/canonical/pebble/client"
+	"gopkg.in/check.v1"
+)
+
+func (cs *clientSuite) TestAddLayer(c *check.C) {
+	for _, combine := range []bool{false, true} {
+		cs.rsp = `{
+		"type": "sync",
+		"status-code": 200,
+		"result": true
+	}`
+		layerYAML := `
+services:
+ foo:
+  override: replace
+  command: cmd
+`[1:]
+		err := cs.cli.AddLayer(&client.AddLayerOptions{
+			Combine:   combine,
+			Label:     "foo",
+			LayerData: []byte(layerYAML),
+		})
+		c.Assert(err, check.IsNil)
+		c.Check(cs.req.Method, check.Equals, "POST")
+		c.Check(cs.req.URL.Path, check.Equals, "/v1/layers")
+		c.Check(cs.req.URL.Query(), check.HasLen, 0)
+		var body map[string]interface{}
+		c.Assert(json.NewDecoder(cs.req.Body).Decode(&body), check.IsNil)
+		c.Assert(body, check.DeepEquals, map[string]interface{}{
+			"action":  "add",
+			"combine": combine,
+			"label":   "foo",
+			"format":  "yaml",
+			"layer":   layerYAML,
+		})
+	}
+}
+
+func (cs *clientSuite) TestPlanData(c *check.C) {
+	cs.rsp = `{
+		"type": "sync",
+		"status-code": 200,
+		"result": "services:\n foo:\n  override: replace\n  command: cmd\n"
+	}`
+	data, err := cs.cli.PlanData(&client.PlanDataOptions{})
+	c.Assert(err, check.IsNil)
+	c.Check(cs.req.Method, check.Equals, "GET")
+	c.Check(cs.req.URL.Path, check.Equals, "/v1/plan")
+	c.Check(cs.req.URL.Query(), check.DeepEquals, url.Values{"format": []string{"yaml"}})
+	c.Assert(string(data), check.Equals, `
+services:
+ foo:
+  override: replace
+  command: cmd
```

Comment:
Just FYI, there's a function inside the setup (now plan) package that reformats yaml so that we can have it properly aligned in cases like this. If it's generally useful, we might move it to the strutil package.

Reply:
Noted. In this case I just "properly aligned" it in the test (4 spaces like `yaml.Marshal` does) -- I think it's clearer in this case  just write explicitly in the test the bytes you want to see.